### PR TITLE
Adding Switch.System.Runtime.Serialization.UseNewMaxArraySize

### PR DESCRIPTION
**Pull request by tahazayed**
## Summary

Add missing AppContextSwitchOverrides: Switch.System.Runtime.Serialization.UseNewMaxArraySize

Related to <https://github.com/dotnet/corefx/issues/16991>

source:https://aloiskraus.wordpress.com/2017/04/23/the-definitive-serialization-performance-guide/

### CODE REVIEW

File: docs/framework/configure-apps/file-schema/runtime/appcontextswitchoverrides-element.md [hunk 1]
```diff
@@ -113,6 +113,7 @@ Defines one or more switches used by the <xref:System.AppContext> class to provi
 |`Switch.UseLegacyAccessibilityFeatures`|Controls whether accessibility features available starting with the .NET Framework 4.7.1 are enabled or disabled. | .NET Framework 4.7.1 |
 |`Switch.UseLegacyAccessibilityFeatures.2`|Controls whether accessibility features available in the .NET Framework 4.7.2 are enabled (`false`) or disabled (`true`). If `true`, `Switch.UseLegacyAccessibilityFeatures` must also be `true` to enable .NET Framework 4.7.1 accessibility features.|.NET Framework 4.7.2|
 |`System.Xml.`<br /><br /> `IgnoreEmptyKeySequences`|Controls whether empty key sequences in compound keys are ignored by XSD schema validation. For more information, see [Mitigation: XML Schema Validation](../../../migration-guide/mitigation-xml-schema-validation.md).|.NET Framework 4.6|  
+|`Switch.System.Runtime.Serialization.UseNewMaxArraySize`|Use this switch to make BinaryFormatter fast with large object graphs starting with .NET 4.7.2|.NET Framework 4.7.2|
```
Comment (rpetrusha) [inline docs/framework/configure-apps/file-schema/runtime/appcontextswitchoverrides-element.md]: Entries in the table are in alphabetical order. This switch should follow `Switch.System.Runtime.Serialization.DoNotUseTimeZoneInfo`.

I've also proposed an alternate wording:
```suggestion
|`Switch.System.Runtime.Serialization.UseNewMaxArraySize`|Controls whether <xref:System.Runtime.Serialization.ObjectManager?displayProperty=nameWithType> uses a larger array size during object serialization and deserialization. Set this switch to `true` to improve the performance of serialization and deserialization of large object graphs by types such as <xref:System.Runtime.Serialization.Formatters.Binary.BinaryFormatter>. |.NET Framework 4.7.2|
```

**Review by rpetrusha — CHANGES_REQUESTED** [review]
Thank you so much for contributing to the dotnet/docs repo and for documenting a configuration switch that we'd overlooked, @tahazayed. I've suggested an alternate wording for the entry. In addition, the switches in the table appear in alphabetical order.

File: docs/framework/configure-apps/file-schema/runtime/appcontextswitchoverrides-element.md (hunk 1 above)
Comment (tahazayed) [reply docs/framework/configure-apps/file-schema/runtime/appcontextswitchoverrides-element.md]: I am working on a program has more features than AutoCAD an it is written in C# and it has a severe performance issues. This switch saved about 6 months of work and I want to share it with others

**Review by rpetrusha — APPROVED** [review]

**Comment by rpetrusha** [comment]
Thank you for making the additional changes, @tahazayed. And again, thank you so much for submitting this PR in the first place and documenting a switch that we'd overlooked. I'll merge your PR now. Your contribution should be live on docs.microsoft.com in the next day or two.

**Pull request merged**